server/src/main/java/org/opensearch/plugins/PluginInfo.java

```diff
@@ -209,6 +217,12 @@ public PluginInfo(final StreamInput in) throws IOException {
         this.customFolderName = in.readString();
         this.extendedPlugins = in.readStringList();
         this.hasNativeController = in.readBoolean();
+        this.optionalExtendedPlugins = new HashSet<>();
```

Comment:
```
if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
    this.optionalExtendedPlugins = in.readStringList();
}
```

Reply:
Please also add it to `writeTo`, to keep serde consistent

Reply:
Added, but why 3_0_0 and not 2_19_9?

Reply:
> Added, but why 3_0_0 and not 2_19_0?

2.19 is fine, sometimes BWC tests may fail (since the change is not in 2.x branch)

Reply:
Changed to V_3_0_0 because of failing tests...after backport would there be a follow-up PR to update this value to 2_19_0?

Reply:
Yep, thank you